vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go

```diff
@@ -82,53 +103,75 @@ func getProportionalLimit(originalLimit, originalRequest, recommendedRequest *re
 	}
 
 	// Input and output milli values should fit in int64 but intermediate values might be bigger.
-	originalMilliRequest := big.NewInt(originalRequest.MilliValue())
-	originalMilliLimit := big.NewInt(originalLimit.MilliValue())
-	recommendedMilliRequest := big.NewInt(recommendedRequest.MilliValue())
-	var recommendedMilliLimit big.Int
-	recommendedMilliLimit.Mul(recommendedMilliRequest, originalMilliLimit)
-	recommendedMilliLimit.Div(&recommendedMilliLimit, originalMilliRequest)
-	if recommendedMilliLimit.IsInt64() {
-		return resource.NewMilliQuantity(recommendedMilliLimit.Int64(), recommendedRequest.Format), false
-	}
-	return resource.NewMilliQuantity(math.MaxInt64, recommendedRequest.Format), true
+	return scaleQuantityProportionally( /*scaledQuantity=*/ originalLimit /*scaleBase=*/, originalRequest /*scaleResult=*/, recommendedRequest)
+}
+
+func proportionallyCapLimitToMax(recommendedRequest, recommendedLimit, maxLimit *resource.Quantity) (request, limit *resource.Quantity) {
+	if recommendedLimit == nil || maxLimit == nil || maxLimit.IsZero() {
+		return recommendedRequest, recommendedLimit
+	}
+	if recommendedLimit.Cmp(*maxLimit) <= 0 {
+		return recommendedRequest, recommendedLimit
+	}
+	scaledRequest, _ := scaleQuantityProportionally(recommendedRequest, recommendedLimit, maxLimit)
+	return scaledRequest, maxLimit
+}
+
+func proportionallyCapLimitsToMax(recommendedRequests v1.ResourceList, cpuLimit, memLimit, maxCpuLimit, maxMemLimit *resource.Quantity) ContainerResources {
```

Comment:
I'd call it proportionallyCapResourcesToMaxLimit, since it also caps request (I was suprised)

Reply:
This logic actually changes the recommended request, but this is not included in the updater - updater and admission controller will disagree about the request - we can get into an infinite updating loop.